src/khepri_machine.erl

```diff
@@ -166,10 +175,13 @@
          %% Added in machine version 1.
          dedups = #{} :: khepri_machine:dedups_map()}).
 
--opaque state_v1() :: #khepri_machine{}.
+-type state_v1() :: #khepri_machine{tree :: khepri_tree:tree_v0()}.
 %% State of this Ra state machine, version 1.
 
--type state() :: state_v1() | khepri_machine_v0:state().
+-type state_v2() :: #khepri_machine{tree :: khepri_tree:tree_v1()}.
+%% State of this Ra state machine, version 2.
```

Comment:
After putting more thoughts, I think we can use a single `state_v1()` opaque type. Both versions are covered by the underlying `tree_v0()` and `tree_v1()` types now.